doc/content/source/problems/NekRSProblem.md

```diff
@@ -140,22 +145,77 @@ the last time it was sent to NekRS). A means by which to reduce some of these
 In the `TO_EXTERNAL_APP` data transfer, [MooseVariables](https://mooseframework.inl.gov/source/variables/MooseVariable.html)
  are read from the
 [NekRSMesh](/mesh/NekRSMesh.md) mesh mirror and interpolated onto the NekRS
-[!ac](GLL) points corresponding to each node using Vandermonde matrices.
-The data transfers going into NekRS are determined based on how the
+[!ac](GLL) points corresponding to each node using [Vandermonde matrices](vandermonde.md).
+The specific data transfers going into NekRS are determined based on how the
 `NekRSMesh` was constructed, i.e. whether boundary and/or volume coupling is used.
 
+Data is "sent" into NekRS by writing into the `nrs->usrwrk` scratch space array,
+which NekRS makes available within the boundary condition functions in the `.oudf` file
+(on device, this array is technically called the `nrs->o_usrwrk` array).
+[usrwrk_nrsp] shows the assignment of "slots" in the `nrs->usrwrk` scratch space
+array with quantities written by Cardinal. Because different quantities are written into
+Cardinal depending on the problem setup, if a particular slice is not needed for a
+case, it will just hold zero values. That is, the **order** of the various quantities
```

Comment:
Say I want to pass `flux heat source y_displacement`:
1. Won't we be wasting memory zeros allocated for `x_displacement` in our efforts to keep the order constant for developers?
2. As a user, I would expect y_displacement to be stored at offset #3 (i.e. 2), but that could just be me.
3. We already expect the user to keep their `usrwrk` slices and Cardinal's `usrwrk` slices separate. I don't think it's too bad to pass on the responsibility to keep offsets straight to the user, as long as we're clear in the documentation.
4. For the convenience of developers, we could use an enum to make sure that 
  a) "y_displacement" is a valid input
  b) map y_displacement to a valid offset, that is incremented each time a valid input is detected. 
  That way developers have to call on one enum, which is stored/modified in a single place (say `NekInterface.C`)

Reply:
I agree, this part might be confusing. I don't think there's a super clear winner on our two options (keeping the order the same + wasting memory, or having the order be problem-dependent + not wasting memory). Consider the case where a user is solving a moving mesh problem with a volumetric heat source (like the cases we've been working on). If the user wants to momentarily turn off the heat source part, then they'll need to modify the indexing in the `.oudf` file, which opens up the door to making mistakes if they forget to do this. 

I think my long-term solution will be to figure out how to get `libUDF` to recognize Cardinal source code. Then we could have a function in `NekInterface.C` like `usrwrk_index(field::nekWriteEnum & field)` (like you have on your local  `meshmvmt` branch) that can get called directly in the `.oudf` file so that the user doesn't ever mess up the indexing. Then, we can absolutely go the route of not wasting any memory at all. What do you think?

Reply:
I agree. I know I'll make that same mistake very soon, but as long as the documentation is clear about this, it's a good compromise for now, and I think we're right to lean towards wanting to save memory. Hopefully, it won't be too hard for the user to debug - just based on all these terms likely being different orders of magnitude and having separate physical effects that will become apparent relatively easily ("I turned off my heat source, but the temperature is still going up...").

Reply:
At some point, maybe I should try to create a segfault based around a potential user error in this scenario (reading outside the bounds of currently allocated `usrwrk`), and write a debugging guide based off of that (once we have the HYPRE bug figured out) for the Cardinal documentation.

Reply:
Yeah, the only change that this PR makes is to allow the users to "cut off" allocating members in `usrwrk` that they definitely don't need. In the future if we want to exclusively allocate only what the user needs, I'd prefer to do that as a separate PR, since it'll require some additional testing. At the moment, pretty much everyone using `NekRSProblem` is only using the conjugate heat transfer capabilties, and only need a single slot in `nrs->usrwrk` saved for flux. 

When the mesh solver is in, let's definitely revisit this!